src/callinput.c

```diff
@@ -796,9 +801,15 @@ int callinput(void) {
 	    // <Escape>, clear call input or stop sending.
 	    case ESCAPE: {
 		if (early_started == 0) {
+			if (escape_pressed == 0) {
```

Comment:
Is the point of using `escape_pressed` is to `stoptx()` for the first ESC and `cleanup()` + `clear_display()` for the second?

Reply:
Exactly.

I find the combination of stopping TX and clearing the exchange field to be rather irritating.  Other programs I've used never did that, in fact they only used Escape to stop TX and another key/combination to clear the fields.

For CW this worked very well this past weekend.  I gave up on stopping the voice keyer as per Tom and my discussion on the mailing list.  I still that should be implemented but there is likely a better way than the hack I came up with (not part of this PR).

Reply:
OK, it took some time for me to figure out. :-)
How about putting it more explicitly, something like
```
if (x==ESC) { ++esc_count } else {esc_count=0}

switch(x) {
  case ESC:
      if (esc_count==1) { stop_tx() }
      else if (esc_count>1) { clear... }

```

Would there be a way to avoid code duplication in callinput and getexechange? I mean some keys should behave the same, so it could warrant a common code. Not sure how to do it best.

Reply:
That looks clean to me.

Sometimes there are subtle differences depending on whether a key is input in the call input or exchange field.  I suppose that can be worked out but it would be a big refactor and I'm not sure what the gain would be.

Reply:
Factor out the handling of common keys is something we should go for. 

For the keys which are handled differently we could even add a state variable telling us the actual input field we are in.